### PR TITLE
E2E framework

### PR DESCRIPTION
**Pull request by nckturner**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add e2e test runner based on kubetest2.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Requires:
* https://github.com/kubernetes/kops/pull/13217
* https://github.com/kubernetes-sigs/kubetest2/pull/182

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: charts/aws-cloud-controller-manager/Chart.yaml [hunk 1]
```diff
@@ -2,11 +2,9 @@
 apiVersion: v1
 name: aws-cloud-controller-manager
 description: Installs Cloud Controller Manager for AWS Cloud Provider
-version: 0.0.5
```
Comment (nckturner) [inline charts/aws-cloud-controller-manager/Chart.yaml:5]: Ignore, need to rebase

File: hack/install-e2e-tools.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export GO111MODULE=on
+
+if [[ -z "${KOPS_ROOT}" ]]; then
+    cd $(mktemp -d) > /dev/null
+    git clone https://github.com/kubernetes/kops.git
+    KOPS_ROOT="$(cwd)/kops"
+fi
+
+cd "$HOME/Projects/kubetest2/" > /dev/null
+#go install sigs.k8s.io/kubetest2/...@latest
```
Comment (nckturner) [inline hack/install-e2e-tools.sh:31]: Todo remove

**Review by adityavenneti — APPROVED** [review]
/approve

File: hack/e2e/run.sh [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      #--test=kops \
+      #-- \
+      #--use-built-binaries=true \
+      #--focus-regex="${GINKGO_FOCUS}" \
+      #--parallel 25
+fi
+
+pushd ./tests/e2e
+${GINKGO_BIN} . -p -nodes="${GINKGO_NODES}" -v --focus="${GINKGO_FOCUS}" --skip="${GINKGO_SKIP}" "" -- -kubeconfig="${KUBECONFIG}" -report-dir="${test_run}" -gce-zone="${ZONES%,*}" "${EXPANDED_TEST_EXTRA_FLAGS}"
+popd
```
Comment (wongma7) [inline hack/e2e/run.sh:110]: take down cluster at end?

File: hack/e2e/run.sh [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+if [[ ! -f ${GINKGO_BIN} ]]; then
+  GOBIN=${test_run} go install "github.com/onsi/ginkgo/ginkgo@${GINKGO_VERSION}"
+fi
+
+if [[ -n "${UP}" ]]; then
+    kubetest2 kops \
+      -v 2 \
+      --up \
+      --run-id=${test_run_id} \
+      --cloud-provider=aws \
```
Comment (wongma7) [inline hack/e2e/run.sh:92]: need to build and deploy a ccm as built from the current PR

Comment (olemarkus) [reply hack/e2e/run.sh:92]: The way kops solves this is to upload the image as file and sideload it. May be doable for CCM too.

Comment (nckturner) [reply hack/e2e/run.sh:114]: Spoke with @olemarkus about this possibility and it sound interesting, (especially when using the ECR kubelet image credential provider).  He is going to do some exploratory work but in the meantime we will just push to ECR and expect an ECR repository to be present in the current account.

File: hack/e2e/run.sh [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+echo "Installing ginkgo to ${test_run}"
+GINKGO_BIN="${test_run}/ginkgo"
+if [[ ! -f ${GINKGO_BIN} ]]; then
+  GOBIN=${test_run} go install "github.com/onsi/ginkgo/ginkgo@${GINKGO_VERSION}"
+fi
+
+echo "Building and pushing test driver image to ${IMAGE_NAME}:${IMAGE_TAG}"
+aws ecr get-login-password --region "${REGION}" | docker login --username AWS --password-stdin "${AWS_ACCOUNT_ID}.dkr.ecr.${REGION}.amazonaws.com"
+docker tag "${IMAGE}" "${IMAGE_NAME}:${IMAGE_TAG}"
```
Comment (wongma7) [inline hack/e2e/run.sh:120]: is IMAGE expected to be an arg? no magic allowed, it should be defined at the top with other args

File: tests/e2e/testsuites/specs.go [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+//. "github.com/onsi/ginkgo"
+
+type ServiceDetails struct {
```
Comment (wongma7) [inline tests/e2e/testsuites/specs.go:18]: I don't really understand the organization of these files

example_tester.go
specs.go
testsuites.go

plz shuffle them or add some comments explaining it

File: hack/e2e/run.sh (hunk 5 above)
Comment (wongma7) [reply hack/e2e/run.sh:120]: might need some prefixes to disambiguate IMAGE from IMAGE_NAME and IMAGE_TAG... IMAGE is the image that is assumed t o exist, IMAGE_NAME:IMAGE_TAG is the destination to push it to

Comment (nckturner) [reply hack/e2e/run.sh]: Updated to use `$MAKE_` prefix to differentiate variables passed from Makefile.

**Pull request merged**